src/Host.cpp

```diff
@@ -1276,3 +1283,250 @@ void Host::setWideAmbiguousEAsianGlyphs(const Qt::CheckState state)
+
+    auto json = document.object();
+    if (json.isEmpty()) {
+        return;
+    }
+
+    if (packageMessage == QLatin1String("External.Discord.Status")) {
+        processGMCPDiscordStatus(json);
+    } else if (packageMessage == QLatin1String("External.Discord.Info")) {
+        processGMCPDiscordInfo(json);
+    }
+}
+
+void Host::processGMCPDiscordInfo(const QJsonObject& discordInfo)
+{
+    mudlet* pMudlet = mudlet::self();
+    bool hasInvite = false;
+    auto inviteUrl = discordInfo.value(QStringLiteral("inviteurl"));
+    // Will be of form: "https://discord.gg/#####"
+    if (inviteUrl != QJsonValue::Undefined) {
+        hasInvite = true;
+    }
+
+    bool hasApplicationId = false;
+    bool hasCustomAppID = false;
+    auto appID = discordInfo.value(QStringLiteral("applicationid"));
+    if (appID != QJsonValue::Undefined) {
+        hasApplicationId = true;
+        if (appID.toString() == Discord::mMudletApplicationId) {
+            pMudlet->mDiscord.setApplicationID(this, QString());
+        } else {
+            hasCustomAppID = true;
+            pMudlet->mDiscord.setApplicationID(this, appID.toString());
+        }
+    }
+
+    if (hasInvite) {
+        if (hasCustomAppID) {
+            qDebug() << "Game using a custom Discord server. Invite URL: " << inviteUrl.toString();
+        } else if (hasApplicationId) {
+            qDebug() << "Game using Mudlets Discord server. Invite URL: " << inviteUrl.toString();
+        } else {
+            qDebug() << "Discord invite URL: " << inviteUrl.toString();
+        }
+    } else {
+        if (hasCustomAppID) {
+            qDebug() << "Game is using custom server Discord application ID";
+        } else if (hasApplicationId) {
+            qDebug() << "Game is using Mudlets Discord application ID";
+        }
+    }
+}
+
+void Host::processGMCPDiscordStatus(const QJsonObject& discordInfo)
+{
+    auto pMudlet = mudlet::self();
+    auto gameName = discordInfo.value(QStringLiteral("game"));
+    if (gameName != QJsonValue::Undefined) {
+        QPair<bool, QString> richPresenceSupported = pMudlet->mDiscord.gameIntegrationSupported(getUrl());
+        if (richPresenceSupported.first && pMudlet->mDiscord.usingMudletsDiscordID(this)) {
+            pMudlet->mDiscord.setDetailText(this, tr("Playing %1").arg(richPresenceSupported.second));
+            pMudlet->mDiscord.setLargeImage(this, richPresenceSupported.second);
+            pMudlet->mDiscord.setLargeImageText(this, tr("%1 at %2:%3").arg(richPresenceSupported.second, getUrl(), QString::number(getPort())));
+        } else {
+            // We are using a custom application id, so the top line is
+            // likely to be saying "Playing MudName"
+            if (richPresenceSupported.first) {
+                pMudlet->mDiscord.setDetailText(this, tr("Using Mudlet"));
+                pMudlet->mDiscord.setLargeImageText(this, tr("%1 at %2:%3").arg(richPresenceSupported.second, getUrl(), QString::number(getPort())));
```

Comment:
This tr() needs a comment to explain what %1 %2 %3 mean and how to translate "at"
Example: `tr("%1 at %2:%3", "%1 is user name (really?) and %2:%3 is game server address like: mudlet.org:23")`